pwndbg/search.py

```diff
@@ -33,26 +32,29 @@ def search(searchfor, mappings=None, start=None, end=None,
     Yields:
         An iterator on the address matches
     """
-    value = searchfor
-    size  = None
-
     i = gdb.selected_inferior()
 
     maps = mappings or pwndbg.vmmap.get()
-    hits = []
 
     if end and start:
-        maps = [m for m in maps if start <= m < end]
+        assert end > start, 'Last address to search must be greater then first address'
+        maps = [m for m in maps if start in m or (end-1) in m]
+    elif start:
+        maps = [m for m in maps if start in m]
+    else:
+        maps = [m for m in maps if (end-1) in m]
 
     if executable:
         maps = [m for m in maps if m.execute]
 
     if writable:
         maps = [m for m in maps if m.write]
 
+    search_start, search_end = start, end
+
     for vmmap in maps:
-        start = vmmap.vaddr
-        end   = start + vmmap.memsz
+        start = vmmap.start if search_start < vmmap.start else search_start
+        end   = vmmap.end if search_end > vmmap.end else search_end
```

Comment:
Right now I am not sure if this is needed as there is the `length = end - start` and `if length <= 0: break` below.